### DISCUSSION TITLE
Performance of external sort

### DISCUSSION THREAD

**Discussion by bbuchfink** [duckdb/duckdb#15163] [General]
I have a table with 3 int64 columns and 17 billion rows that I want to sort on the first column. If I run the sort in memory it is very fast. I want to do it externally like this:
```
set threads=64;
set memory_limit='64GB';
select * from my_table order by column0;
```
This runs quickly to 50% progress at which point it starts hanging for several hours running on 1-2 threads, after which I killed it. The temporary disk space use is ~600GB at this point, which seems a lot given that the Duckdb database size is ~200GB. Running v1.1.3 on a Linux AMD EPYC server with 128 cores, 2 TB of memory. The storage is a SSD-based NFS mount. Any chance of improving this?

**Comment by soerenwolfers** [comment]
I'd open a proper issue for this. If possible with a table that you first generate in DuckDB or in Python; alternatively with a dataset that you upload somewhere.

**Reply by soerenwolfers** [reply]
.
